python/cudf/cudf_pandas_tests/test_cudf_pandas.py

```diff
@@ -1421,3 +1422,48 @@ def test_holidays_within_dates(holiday, start, expected):
             utc.localize(xpd.Timestamp(start)),
         )
     ) == [utc.localize(dt) for dt in expected]
+
+
+def test_cudf_pandas_debugging_different_results(monkeypatch):
+    def mock_mean_float(self, *args, **kwargs):
+        return np.float64(1.0)
+
+    with monkeypatch.context() as monkeycontext:
+        monkeycontext.setattr(cudf.Series, "mean", mock_mean_float)
+        monkeycontext.setenv("CUDF_PANDAS_DEBUGGING", "True")
+        s = xpd.Series([1, 2])
+        with pytest.warns(
+            UserWarning,
+            match="The results from cudf and pandas were different.",
+        ):
+            assert s.mean() == np.float64(1.0)
+
+
+def test_cudf_pandas_debugging_pandas_error(monkeypatch):
+    def mock_mean_exception(self, *args, **kwargs):
+        raise Exception()
+
+    with monkeypatch.context() as monkeycontext:
+        monkeycontext.setattr(pd.Series, "mean", mock_mean_exception)
+        monkeycontext.setenv("CUDF_PANDAS_DEBUGGING", "True")
+        s = xpd.Series([1, 2])
+        with pytest.warns(
+            UserWarning,
+            match="The result from pandas could not be computed.",
+        ):
+            assert s.mean() == 1.5
```

Comment:
Looks like tests are failing here. It seems like the monkeypatch from the previous test is not being un-patched?

```python
=================================== FAILURES ===================================
___________________ test_cudf_pandas_debugging_pandas_error ____________________

monkeypatch = <_pytest.monkeypatch.MonkeyPatch object at 0x7f8ba3845a30>

    def test_cudf_pandas_debugging_pandas_error(monkeypatch):
        def mock_mean_exception(self, *args, **kwargs):
            raise Exception()
    
        with monkeypatch.context() as monkeycontext:
            monkeycontext.setattr(pd.Series, "mean", mock_mean_exception)
            monkeycontext.setenv("CUDF_PANDAS_DEBUGGING", "True")
            s = xpd.Series([1, 2])
            with pytest.warns(
                UserWarning,
                match="The result from pandas could not be computed.",
            ):
>               assert s.mean() == 1.5
E               assert 1.0 == 1.5
E                +  where 1.0 = <bound method Series.mean of 0    1\n1    2\ndtype: int64>()
E                +    where <bound method Series.mean of 0    1\n1    2\ndtype: int64> = 0    1\n1    2\ndtype: int64.mean

python/cudf/cudf_pandas_tests/test_cudf_pandas.py:1454: AssertionError
______________________ test_cudf_pandas_debugging_failed _______________________

monkeypatch = <_pytest.monkeypatch.MonkeyPatch object at 0x7f8ba38646a0>

    def test_cudf_pandas_debugging_failed(monkeypatch):
        def mock_mean_none(self, *args, **kwargs):
            return None
    
        with monkeypatch.context() as monkeycontext:
            monkeycontext.setattr(pd.Series, "mean", mock_mean_none)
            monkeycontext.setenv("CUDF_PANDAS_DEBUGGING", "True")
            s = xpd.Series([1, 2])
            with pytest.warns(
                UserWarning,
                match="Pandas debugging mode failed.",
            ):
>               assert s.mean() == 1.5
E               assert 1.0 == 1.5
E                +  where 1.0 = <bound method Series.mean of 0    1\n1    2\ndtype: int64>()
E                +    where <bound method Series.mean of 0    1\n1    2\ndtype: int64> = 0    1\n1    2\ndtype: int64.mean

python/cudf/cudf_pandas_tests/test_cudf_pandas.py:1469: AssertionError
```

Reply:
Yes, I tried explicitly calling `monkeypatch.undo()` but that also failed

Reply:
I applied a workaround (monkeypatching different functions in each test)

Reply:
This workaround feels wrong. We need the monkeypatch to be un-applied correctly... we don't want to just leave it patched. Can you reproduce this problem locally?

Reply:
I think the issue is fixed by patching the `_fsproxy_fast` attr of `_MethodProxy`. I'm going to run tests again when CI networks issues are resolved.

Reply:
ie. 
```diff
+ monkeypatch.setattr(xpd.Series.mean, "_fsproxy_fast", mock_mean_one)
- monkeypatch.setattr(cudf.Series, "mean", mock_mean_one)
```

Reply:
Cool, I think that should work! Thanks for pursuing this. Can you add a comment to explain why you're patching the `_fsproxy_fast`?